components/telemetry-operator/internal/configbuilder/config_builder_test.go

```diff
@@ -0,0 +1,144 @@
+		Spec: telemetryv1alpha1.LogPipelineSpec{
+			Output: telemetryv1alpha1.Output{
+				HTTP: telemetryv1alpha1.HTTPOutput{
+					Dedot: true,
+					Host: telemetryv1alpha1.ValueType{
+						Value: "localhost"}}}}}
+
+	actual := createLuaDedotFilter(logPipeline)
+	require.Equal(t, expected, actual)
+}
+
+func TestCreateLuaDedotFilterWithUndefinedHost(t *testing.T) {
+	logPipeline := &telemetryv1alpha1.LogPipeline{
+		Spec: telemetryv1alpha1.LogPipelineSpec{
+			Output: telemetryv1alpha1.Output{
+				HTTP: telemetryv1alpha1.HTTPOutput{
+					Dedot: true}}}}
+
+	actual := createLuaDedotFilter(logPipeline)
+	require.Equal(t, "", actual)
+}
+
+func TestCreateLuaDedotFilterWithDedotFalse(t *testing.T) {
+	logPipeline := &telemetryv1alpha1.LogPipeline{
+		Spec: telemetryv1alpha1.LogPipelineSpec{
+			Output: telemetryv1alpha1.Output{
+				HTTP: telemetryv1alpha1.HTTPOutput{
+					Dedot: false,
+					Host: telemetryv1alpha1.ValueType{
+						Value: "localhost"}}}}}
+
+	actual := createLuaDedotFilter(logPipeline)
+	require.Equal(t, "", actual)
+}
+
+func TestMergeSectionsConfig(t *testing.T) {
+	expected := `[FILTER]
+    Emitter_Mem_Buf_Limit 10M
+    Emitter_Name          foo
+    Emitter_Storage.type  filesystem
+    Match                 kube.*
+    Name                  rewrite_tag
+    Rule                  $kubernetes['namespace_name'] "^(?!kyma-system$|kyma-integration$|kube-system$|istio-system$).*" foo.$TAG true
+
+[FILTER]
+    match  foo.*
+    name   record_modifier
+    record cluster_identifier ${KUBERNETES_SERVICE_HOST}
+
+[FILTER]
+    match foo.*
+    name  grep
+    regex log aa
+
+[FILTER]
+    call   kubernetes_map_keys
+    match  foo.*
+    name   lua
+    script /fluent-bit/scripts/filter-script.lua
+
+[OUTPUT]
+    allow_duplicated_headers true
+    format                   json
+    host                     localhost
+    match                    foo.*
+    name                     http
+    port                     443
+    storage.total_limit_size 1G
+    tls                      on
+    tls.verify               on
+
+`
+	logPipeline := &telemetryv1alpha1.LogPipeline{
+		Spec: telemetryv1alpha1.LogPipelineSpec{
+			Input: telemetryv1alpha1.Input{
+				Application: telemetryv1alpha1.ApplicationInput{
+					KeepAnnotations: true,
+					DropLabels:      false,
+				},
+			},
+			Filters: []telemetryv1alpha1.Filter{
+				{
+					Custom: `
+						name grep
+						regex log aa
+					`,
+				},
+			},
+			Output: telemetryv1alpha1.Output{
+				HTTP: telemetryv1alpha1.HTTPOutput{
+					Dedot: true,
+					Host: telemetryv1alpha1.ValueType{
+						Value: "localhost",
+					},
+				},
+			},
+		},
+	}
+	logPipeline.Name = "foo"
+	pipelineConfig := PipelineConfig{
+		InputTag:          "kube",
+		MemoryBufferLimit: "10M",
+		StorageType:       "filesystem",
+		FsBufferLimit:     "1G",
+	}
+
+	actual, err := MergeSectionsConfig(logPipeline, pipelineConfig)
+	require.NoError(t, err)
+	require.Equal(t, expected, actual)
+}
```

Comment:
There is no test case for `MergeParsersConfig` or `buildConfigSection`, there also was no test case before.

Reply:
I agree, would love to do an iteration on `MergeParsersConfig` anyways but not here. Think it's too many changes already.